### PR TITLE
Update to gomod with vendor directory, update vendored pkgs

### PR DESCRIPTION
**Pull request by derekcollison**
Biting the bullet since I needed to update jwt dependency and `gvt` is no longer supported or is able to do updates.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: vendor/github.com/nats-io/nuid/LICENSE [hunk 1]
```diff
@@ -1,201 +1,21 @@
-                                 Apache License
```
Comment (kozlovic) [inline vendor/github.com/nats-io/nuid/LICENSE]: This is strange.. looks like it removed Apache license and went back to MIT license?

Comment (derekcollison) [reply vendor/github.com/nats-io/nuid/LICENSE]: hmm let me look I see it too..

Comment (derekcollison) [reply vendor/github.com/nats-io/nuid/LICENSE]: Might not have tagged latest, modules seem to like releases..

**Comment by kozlovic** [comment]
LGTM!

**Pull request merged**